bin-src/lib/getDependentStoryFiles.ts

```diff
@@ -31,6 +21,55 @@ const isUserModule = (mod: Module | Reason) =>
 // already contains forward slashes, because that's what git yields even on Windows.
 const posix = (localPath: string) => localPath.split(path.sep).filter(Boolean).join(path.posix.sep);
 
+// put this in file with same name
+const getExternalChangedDirectDependencies = async (changedFiles) => {
+  const dependenciesChanged = [];
+  // const changedPackageFiles = changedFiles.filter(changedFile => /(^package\.json$)|(\/package\.json$)/.test(changedFile))
+  const changedPackageFiles = ['package.json'];
+  try {
+    await Promise.all(
+      changedPackageFiles.map(async (packageFilePath) => {
+        // do same build trick to get old commit
+        const changes = await execGitCommand(`git --no-pager diff ${packageFilePath}`);
+        const filez = parseDiff(changes);
+        filez[0].chunks.forEach((chunk) => {
+          const theChanges = chunk.changes.filter((item) => item.type === 'add');
+          // make sure this change is a proper change... should be easy enough...
+          // honestly just pull out the package name
+          // (later check that it is in proper format, maybe just using semver)
+          const refinedChanges = theChanges.map((aChange) =>
+            aChange.content.split('+')[1].split(':')[0].trim().replace(/"/g, '')
+          );
```

Comment:
I feel like there might be some ways this could go wrong (what if the dependency goes over two lines somehow, etc)?

Would we be better of actually just getting the before and after JSON of the `package.json` and comparing directly?

Reply:
Thanks for the heads-up, will give this a look.

Reply:
@tmeasday thanks for this point, just to make sure I'm on the same page -- you're suggesting to do something like `git show` on the file at a given commit, and then diff the keys of the dependencies (and devdeps/peerdeps) to find the difference? Sounds good, just wanted to make sure I understood your point.

Reply:
Yep, that's it! I realised as I wrote that comment that that approach might be a lot easier than trying to parse + apply the diff.